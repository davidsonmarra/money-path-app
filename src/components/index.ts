export { default as Button, ButtonType } from 'src/components/button';
export {
  default as Divider,
  DividerSize,
  DividerType,
} from 'src/components/divider';
export { default as Feedback, FeedbackType } from 'src/components/feedback';
export { default as Header } from 'src/components/header';
export { default as Input } from 'src/components/input';
export { default as TextArea } from 'src/components/text-area';
export { default as ListItem } from 'src/components/list-item';
export { default as ProfileHeader } from 'src/components/profile-header';
export { default as Spacer } from 'src/components/spacer';
export { default as Text, TextType } from 'src/components/text';
export { default as IconRound } from 'src/components/icon-round';
export { default as StorybookView } from 'src/components/storybook-view';
export { default as SelectionGroup } from 'src/components/selection-group';
