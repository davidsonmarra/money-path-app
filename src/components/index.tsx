export {default as Button, ButtonType} from 'src/components/button';
export {default as Text, TextType} from 'src/components/text';
export {default as StorybookView} from 'src/components/storybook-view';
