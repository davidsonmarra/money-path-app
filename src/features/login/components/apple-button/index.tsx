import React from 'react';
import { TouchableOpacityProps } from 'react-native';
import Button from 'src/components/button';
import { AppleIcon } from 'src/assets/icons';
import getStyles from 'src/features/login/components/apple-button/styles';

type Props = TouchableOpacityProps;

const AppleButton = ({ onPress }: Props) => {
  return (
    <Button
      onPress={onPress}
      leftIcon={<AppleIcon size={28} />}
      text="Entrar com Apple"
      style={getStyles().container}
    />
  );
};

export default AppleButton;
