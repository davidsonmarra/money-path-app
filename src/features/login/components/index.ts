export {default as AppleButton} from 'src/features/login/components/apple-button';
export {default as GoogleButton} from 'src/features/login/components/google-button';
