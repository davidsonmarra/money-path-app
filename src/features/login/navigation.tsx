import React from 'react';
import {createStackNavigator} from '@react-navigation/stack';
import LoginScreen from 'src/features/login/screens/login';

const {Navigator, Screen} = createStackNavigator();

const LoginStack = () => {
  return (
    <Navigator
      screenOptions={{
        headerShown: false,
      }}>
      <Screen name="LoginScreen" component={LoginScreen} />
    </Navigator>
  );
};

export default LoginStack;
