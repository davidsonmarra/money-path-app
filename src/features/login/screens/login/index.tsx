import React from 'react';
import auth, {firebase} from '@react-native-firebase/auth';
import {GoogleSignin} from '@react-native-google-signin/google-signin';
import LoginContainer from 'src/features/login/screens/login/ui';

async function onGoogleButtonPress() {
  await GoogleSignin.hasPlayServices({showPlayServicesUpdateDialog: true});
  const signInResult = await GoogleSignin.signIn();

  const idToken = signInResult.data?.idToken;

  if (!idToken) {
    throw new Error('No ID token found');
  }

  const googleCredential = auth.GoogleAuthProvider.credential(idToken);

  await firebase.auth().signInWithCredential(googleCredential);
}

const LoginScreen = () => {
  const handleLoginWithApple = () => {
    console.log('Login with Apple');
  };

  return (
    <LoginContainer
      loginWithApple={handleLoginWithApple}
      loginWithGoogle={onGoogleButtonPress}
    />
  );
};

export default LoginScreen;
