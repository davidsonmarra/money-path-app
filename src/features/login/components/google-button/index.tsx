import React from 'react';
import {TouchableOpacityProps} from 'react-native';
import {Button, ButtonType} from 'src/components';
import {GoogleIcon} from 'src/assets/icons';
import getStyles from 'src/features/login/components/google-button/styles';

type Props = TouchableOpacityProps;

const GoogleButton = ({onPress}: Props) => {
  return (
    <Button
      onPress={onPress}
      leftIcon={<GoogleIcon size={28} />}
      text="Entrar com Google"
      style={getStyles().container}
      type={ButtonType.secondary}
    />
  );
};

export default GoogleButton;
