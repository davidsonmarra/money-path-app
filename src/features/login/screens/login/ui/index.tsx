import React from 'react';
import {Platform, View} from 'react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import {Text, TextType, Spacer} from 'src/components';
import {MoneyPathIcon} from 'src/assets/icons';
import {AppleButton, GoogleButton} from 'src/features/login/components';
import useStyles from 'src/features/login/screens/login/ui/styles';

interface Props {
  loginWithGoogle: () => void;
  loginWithApple: () => void;
}

const renderButtonsAndroid = (loginWithGoogle: () => void) => (
  <GoogleButton onPress={loginWithGoogle} />
);

const renderButtonsIOS = (
  loginWithApple: () => void,
  loginWithGoogle: () => void,
) => (
  <>
    <AppleButton onPress={loginWithApple} />
    <Spacer height={8} />
    <GoogleButton onPress={loginWithGoogle} />
  </>
);

const LoginContainer = ({loginWithApple, loginWithGoogle}: Props) => {
  return (
    <SafeAreaView style={useStyles().container}>
      <View style={useStyles().logoContainer}>
        <MoneyPathIcon size={72} />
      </View>
      <Text type={TextType.textMediumRegular}>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec
        vestibulum, urna nec lacinia. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit. Donec vestibulum, urna nec lacinia.
      </Text>
      <View>
        {Platform.select({
          ios: renderButtonsIOS(loginWithApple, loginWithGoogle),
          android: renderButtonsAndroid(loginWithGoogle),
        })}
      </View>
    </SafeAreaView>
  );
};

export default LoginContainer;
