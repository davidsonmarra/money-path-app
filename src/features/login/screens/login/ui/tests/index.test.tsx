import React from 'react';
import {render} from 'src/configs/test-utils';
import {AppleButton, GoogleButton} from 'src/features/login/components';
import LoginContainer from 'src/features/login/screens/login/ui';

const containerInstance = (props: any) => render(<LoginContainer {...props} />);

describe('LoginContainer', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const container = containerInstance({}).toJSON();
    expect(container).toMatchSnapshot();
  });

  it('should call loginWithGoogle', () => {
    const loginWithGoogle = jest.fn();
    const container = containerInstance({loginWithGoogle}).root;
    container.findByType(GoogleButton).props.onPress();
    expect(loginWithGoogle).toHaveBeenCalled();
  });

  it('should call loginWithApple', () => {
    const loginWithApple = jest.fn();
    const container = containerInstance({loginWithApple}).root;
    container.findByType(AppleButton).props.onPress();
    expect(loginWithApple).toHaveBeenCalled();
  });
});
