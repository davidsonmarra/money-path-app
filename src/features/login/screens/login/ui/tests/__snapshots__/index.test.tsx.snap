// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginContainer should render correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    style={
      {
        "backgroundColor": "#232E38",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "padding": 24,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={72}
        bbWidth={72}
        fill="#EAC43D"
        focusable={false}
        height={72}
        meetOrSlice={0}
        minX={0}
        minY={-960}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 72,
              "width": 72,
            },
          ]
        }
        testID="icon-moneyPath"
        vbHeight={960}
        vbWidth={960}
        width={72}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4293575741,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M444-200h70v-50q50-9 86-39t36-89q0-42-24-77t-96-61q-60-20-83-35t-23-41q0-26 18.5-41t53.5-15q32 0 50 15.5t26 38.5l64-26q-11-35-40.5-61T516-710v-50h-70v50q-50 11-78 44t-28 74q0 47 27.5 76t86.5 50q63 23 87.5 41t24.5 47q0 33-23.5 48.5T486-314q-33 0-58.5-20.5T390-396l-66 26q14 48 43.5 77.5T444-252v52zm36 120q-83 0-156-31.5T197-197q-54-54-85.5-127T80-480q0-83 31.5-156T197-763q54-54 127-85.5T480-880q83 0 156 31.5T763-763q54 54 85.5 127T880-480q0 83-31.5 156T763-197q-54 54-127 85.5T480-80zm0-80q134 0 227-93t93-227q0-134-93-227t-227-93q-134 0-227 93t-93 227q0 134 93 227t227 93zm0-320z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        [
          {
            "color": "#D9D9D9",
            "fontSize": 16,
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec vestibulum, urna nec lacinia. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec vestibulum, urna nec lacinia.
    </Text>
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 12,
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingHorizontal": 24,
            "paddingVertical": 12,
            "width": "100%",
          }
        }
      >
        <View>
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={28}
            bbWidth={28}
            fill="#FFF"
            focusable={false}
            height={28}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 28,
                  "width": 28,
                },
              ]
            }
            testID="icon-apple"
            vbHeight={22.773}
            vbWidth={22.773}
            width={28}
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M15.769 0h.162c.13 1.606-.483 2.806-1.228 3.675-.731.863-1.732 1.7-3.351 1.573-.108-1.583.506-2.694 1.25-3.561C13.292.879 14.557.16 15.769 0zM20.67 16.716v.045c-.455 1.378-1.104 2.559-1.896 3.655-.723.995-1.609 2.334-3.191 2.334-1.367 0-2.275-.879-3.676-.903-1.482-.024-2.297.735-3.652.926h-.462c-.995-.144-1.798-.932-2.383-1.642-1.725-2.098-3.058-4.808-3.306-8.276v-1.019c.105-2.482 1.311-4.5 2.914-5.478.846-.52 2.009-.963 3.304-.765.555.086 1.122.276 1.619.464.471.181 1.06.502 1.618.485.378-.011.754-.208 1.135-.347 1.116-.403 2.21-.865 3.652-.648 1.733.262 2.963 1.032 3.723 2.22-1.466.933-2.625 2.339-2.427 4.74.176 2.181 1.444 3.457 3.028 4.209z"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          style={
            [
              {
                "color": "#FFF",
                "fontSize": 16,
                "fontWeight": "bold",
              },
              undefined,
            ]
          }
        >
          Entrar com Apple
        </Text>
        <View />
      </View>
      <View
        style={
          {
            "height": 8,
            "width": 0,
          }
        }
        testID="spacer"
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 12,
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingHorizontal": 24,
            "paddingVertical": 12,
            "width": "100%",
          }
        }
        testID="btn-google"
      >
        <View>
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={28}
            bbWidth={28}
            focusable={false}
            height={28}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 28,
                  "width": 28,
                },
              ]
            }
            testID="icon-google"
            vbHeight={48}
            vbWidth={48}
            width={28}
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            >
              <RNSVGPath
                d="M43.611 20.083H42V20H24v8h11.303c-1.649 4.657-6.08 8-11.303 8-6.627 0-12-5.373-12-12s5.373-12 12-12c3.059 0 5.842 1.154 7.961 3.039l5.657-5.657C34.046 6.053 29.268 4 24 4 12.955 4 4 12.955 4 24s8.955 20 20 20 20-8.955 20-20c0-1.341-.138-2.65-.389-3.917z"
                fill={
                  {
                    "payload": 4294951175,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              />
              <RNSVGPath
                d="M6.306 14.691l6.571 4.819C14.655 15.108 18.961 12 24 12c3.059 0 5.842 1.154 7.961 3.039l5.657-5.657C34.046 6.053 29.268 4 24 4 16.318 4 9.656 8.337 6.306 14.691z"
                fill={
                  {
                    "payload": 4294917376,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              />
              <RNSVGPath
                d="M24 44c5.166 0 9.86-1.977 13.409-5.192l-6.19-5.238A11.91 11.91 0 0124 36c-5.202 0-9.619-3.317-11.283-7.946l-6.522 5.025C9.505 39.556 16.227 44 24 44z"
                fill={
                  {
                    "payload": 4283215696,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              />
              <RNSVGPath
                d="M43.611 20.083H42V20H24v8h11.303a12.04 12.04 0 01-4.087 5.571l.003-.002 6.19 5.238C36.971 39.205 44 34 44 24c0-1.341-.138-2.65-.389-3.917z"
                fill={
                  {
                    "payload": 4279858898,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          style={
            [
              {
                "color": "#2A5C99",
                "fontSize": 16,
                "fontWeight": "bold",
              },
              undefined,
            ]
          }
        >
          Entrar com Google
        </Text>
        <View />
      </View>
    </View>
  </RNCSafeAreaView>
</RNCSafeAreaView>
`;
